### PR TITLE
Improve matching of overloaded functions

### LINKED ISSUE #807: Improve matching of overloaded functions
## Description

If overloaded functions have concrete values in signature, the compiler should be able to determine which overloaded function to call. Currently, the compiler fails with `"x" conflicts with another symbol`. This error should be thrown only if one of the overloaded functions is generic.

## Environment

- Compiler version: develop
- Operating system: Ubuntu 18.04 LTS

## Steps to Reproduce

```zokrates
def test() -> field[2]:
    return [1, 2]

def test() -> field[3]:
    return [1, 2, 3]

def main():
    field[2] t = test() // we should know here which function to pick
    return
```

### PR DESCRIPTION
**Pull request by dark64**
Solves https://github.com/Zokrates/ZoKrates/issues/807

### CODE REVIEW

**Comment by dark64** [comment]
@Schaeff is changelog required for this stuff?

**Comment by Schaeff** [comment]
I think so, as this changes behaviour that users would notice. Maybe more than a fix it's an improvement.

File: zokrates_core/src/typed_absy/types.rs [hunk 1]
```diff
@@ -17,22 +17,27 @@ pub enum Constant<'ast> {
...
 }
 
 impl<'ast> Ord for Constant<'ast> {
-    fn cmp(&self, _: &Self) -> std::cmp::Ordering {
-        std::cmp::Ordering::Equal
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        match (self, other) {
+            (Constant::Concrete(a), Constant::Concrete(b)) => a.cmp(b),
+            _ => std::cmp::Ordering::Equal,
+        }
```
Comment (Schaeff) [inline zokrates_core/src/typed_absy/types.rs:40]: The Hash implementation further down should also be updated or removed for consistency.

**Comment by Schaeff** [comment]
I think #826 ends up replacing this, just need to make sure the tests introduced here are covered there.

**Comment by Schaeff** [comment]
This can be closed

**Pull request closed without merging**